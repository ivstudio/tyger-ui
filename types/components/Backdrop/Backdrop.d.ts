/// <reference types="react" />
declare type TZIndex = string;
interface IBackdrop {
    children?: React.ReactNode | React.ReactNode[];
    onClick?: () => void;
    open?: boolean;
    duration?: number;
    base?: boolean;
    key?: string;
    transparent?: boolean;
    zindex?: TZIndex;
}
export declare const BackdropRoot: import("styled-components").StyledComponent<import("framer-motion").ForwardRefComponent<HTMLDivElement, import("framer-motion").HTMLMotionProps<"div">>, import("styled-components").DefaultTheme, {
    color: string;
    zindex?: string | undefined;
}, never>;
declare const Backdrop: ({ key, children, onClick, open, duration, zindex, transparent, }: IBackdrop) => JSX.Element;
export default Backdrop;
