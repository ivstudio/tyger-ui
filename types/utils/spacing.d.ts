export declare const spacing: {
    readonly '0': "0px";
    readonly '2': "2px";
    readonly '4': "4px";
    readonly '8': "8px";
    readonly '12': "12px";
    readonly '16': "16px";
    readonly '24': "24px";
    readonly '32': "32px";
    readonly '40': "40px";
    readonly '48': "48px";
    readonly '64': "64px";
    readonly '80': "80px";
    readonly '96': "96px";
    readonly '160': "160px";
};
export declare type TSpacing = keyof typeof spacing;
