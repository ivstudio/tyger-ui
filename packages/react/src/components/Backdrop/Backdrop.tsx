'use client';
import { AnimatePresence, motion } from 'framer-motion';
import styled from 'styled-components';

import { flexCenterAll } from '../../styles';
import { BackdropProps } from './Backdrop.types';

const dark65 = 'rgba(0, 0, 0, 0.5)';

export const BackdropRoot = styled(motion.div)<{
    color: string;
    zIndex?: string;
}>`
    position: fixed;
    inset: 0px;
    width: 100%;
    ${flexCenterAll}
    background: ${({ color }) => color};
    z-index: ${({ zIndex }) => zIndex};
    -webkit-tap-highlight-color: transparent;
`;

const Backdrop = ({
    key = 'backdrop',
    children,
    onClick,
    open = false,
    duration = 0.15,
    zIndex = '1',
    transparent,
}: BackdropProps) => {
    return (
        <AnimatePresence>
            {open ? (
                <BackdropRoot
                    key={key}
                    data-testid="backdrop"
                    onClick={onClick}
                    initial={{ opacity: 0 }}
                    animate={{ opacity: 1 }}
                    exit={{ opacity: 0 }}
                    transition={{ duration }}
                    color={transparent ? 'transparent' : dark65}
                    zIndex={zIndex}
                >
                    {children && children}
                </BackdropRoot>
            ) : null}
        </AnimatePresence>
    );
};

export default Backdrop;
