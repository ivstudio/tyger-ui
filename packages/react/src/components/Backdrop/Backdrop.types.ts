export interface BackdropProps {
    children?: React.ReactNode | React.ReactNode[];
    onClick?: () => void;
    open?: boolean;
    duration?: number;
    base?: boolean;
    key?: string;
    transparent?: boolean;
    zIndex?: string;
}
