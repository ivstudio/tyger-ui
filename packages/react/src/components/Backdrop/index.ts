'use client';
export { default } from './Backdrop';
