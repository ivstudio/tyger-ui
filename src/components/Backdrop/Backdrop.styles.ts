import { motion } from 'framer-motion';
import styled from 'styled-components';

export const BackdropRoot = styled(motion.div)<{
    color: string;
    zindex?: number;
}>`
    position: fixed;
    left: 0;
    bottom: 0;
    top: 0;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${({ color }) => color};
    z-index: ${({ zindex }) => zindex};
`;
