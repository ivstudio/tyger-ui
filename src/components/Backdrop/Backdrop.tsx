import { AnimatePresence, motion } from 'framer-motion';
import styled from 'styled-components';

interface IBackdrop {
    children?: React.ReactNode | React.ReactNode[];
    onClick?: () => void;
    open?: boolean;
    duration?: number;
    base?: boolean;
    key?: string;
    transparent?: boolean;
    zindex?: number;
}

export const BackdropRoot = styled(motion.div)<{
    color: string;
    zindex?: number;
}>`
    position: fixed;
    left: 0;
    bottom: 0;
    top: 0;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${({ color }) => color};
    z-index: ${({ zindex }) => zindex};
`;

const dark65 = 'rgba(0, 0, 0, 0.65)';

const Backdrop = ({
    key = 'backdrop',
    children,
    onClick,
    open = false,
    duration = 0.15,
    zindex,
    transparent,
}: IBackdrop) => {
    return (
        <AnimatePresence>
            {open ? (
                <BackdropRoot
                    key={key}
                    onClick={onClick}
                    initial={{ opacity: 0 }}
                    animate={{ opacity: 1 }}
                    exit={{ opacity: 0 }}
                    transition={{ duration }}
                    color={transparent ? 'transparent' : dark65}
                    zindex={zindex}
                >
                    {children}
                </BackdropRoot>
            ) : null}
        </AnimatePresence>
    );
};

export default Backdrop;
