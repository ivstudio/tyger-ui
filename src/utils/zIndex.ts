export const zindex = {
    mobileStepper: 1000,
    appBar: 1100,
    drawer: 1200,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
};
export type TZIndex = keyof typeof zindex;
