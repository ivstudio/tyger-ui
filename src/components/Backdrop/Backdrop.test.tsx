import React from 'react';
import Backdrop from './Backdrop';
import { render, fireEvent, screen } from '../../../test/utils/mockProvider';

const clickHandler = jest.fn();

describe('Backdrop', () => {
    it('renders Backdrop', () => {
        const { container } = render(<Backdrop open={true} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('handles click event', () => {
        render(
            <Backdrop onClick={clickHandler} open={true}>
                <p>Lorem ipsum dolor sit amet.</p>
            </Backdrop>
        );
        const content = screen.getByText('Lorem ipsum dolor sit amet.');
        expect(content).toBeInTheDocument();

        fireEvent.click(screen.getByTestId('backdrop'));
        expect(clickHandler).toHaveBeenCalledTimes(1);
    });

    it('hides child component', () => {
        render(
            <Backdrop onClick={clickHandler} open={false}>
                <p>Lorem ipsum dolor sit amet.</p>
            </Backdrop>
        );

        const content = screen.queryByText('Lorem ipsum dolor sit amet.');
        expect(content).not.toBeInTheDocument();
    });
});
