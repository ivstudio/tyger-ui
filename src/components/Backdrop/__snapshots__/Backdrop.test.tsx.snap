// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop renders Backdrop correctly 1`] = `
<div
  class="sc-bczRLJ etZFho"
  color="rgba(0, 0, 0, 0.65)"
  data-testid="backdrop"
  style="opacity: 0;"
/>
`;
